File: implementations/hannoi_gccgo_rec/hannoi.go
package main

import "fmt"
import "os"
import "strconv"

func main() {
	disks := 5
	var err error
	if len(os.Args) > 1 {
		disks, err = strconv.Atoi(os.Args[1])
		if err != nil {
			fmt.Println(err)
		}
	}
	move(disks, 1, 2, 3)
}

func move(n, from, to, via int) {

	if n > 0 {
		move(n-1, from, to, via)
		//fmt.Println("Move disk ", n, " from ", from, " to ", to)
		move(n-1, via, from, to)
	}
}

File: implementations/hannoi_rust_rec/hannoi.rs
use std::env;

fn move_(disk:i32, from : i32 , to : i32 , via :i32  ) 
{
    if disk >0 
    {
        move_ (disk -1 , from , via, to ); 
        //println!("Move disk from pole {} to pole {}", from, to);
        move_ (disk -1 , via, to , from); 
    }
}

fn main( )
 {
     let args : Vec<String> = env::args().collect(); 
     let disks = if args.len() > 1  {args[1].parse::<i32>().unwrap()} else  { 5 } ; 
     move_(disks,1,2,3)
 }

